dockerplugin.py

```diff
@@ -185,15 +186,28 @@ def read_cpu_stats(container, dimensions, stats, t):
 
     # CPU Percentage based on calculateCPUPercent Docker method
     # https://github.com/docker/docker/blob/master/api/client/stats.go
+    cpu_percent = get_cpu_percent(stats)
+    emit(container, dimensions, 'cpu.percent', [cpu_percent], t=t)
+
+
+def get_cpu_percent(stats):
     cpu_percent = 0.0
+    cpu_usage = stats['cpu_stats']['cpu_usage']
     if 'precpu_stats' in stats:
         precpu_stats = stats['precpu_stats']
         precpu_usage = precpu_stats['cpu_usage']
+        percpu = cpu_usage['percpu_usage']
         cpu_delta = cpu_usage['total_usage'] - precpu_usage['total_usage']
-        system_delta = system_cpu_usage - precpu_stats['system_cpu_usage']
-        if system_delta > 0 and cpu_delta > 0:
-            cpu_percent = 100.0 * cpu_delta / system_delta * len(percpu)
-    emit(container, dimensions, "cpu.percent", [cpu_percent], t=t)
+        # Sometimes system_cpu_usage is not in cpu_stats (when there's load)
+        if 'system_cpu_usage' in stats['cpu_stats']:
+            system_cpu_usage = stats['cpu_stats']['system_cpu_usage']
+            if 'system_cpu_usage' in precpu_stats:
+                pre_system_cpu_usage = precpu_stats['system_cpu_usage']
+                system_delta = system_cpu_usage - pre_system_cpu_usage
+                if system_delta > 0 and cpu_delta > 0:
+                    cpu_percent = cpu_delta / system_delta * len(percpu)
```

Comment:
Actually you should keep this as `100.0 * cpu_delta / system_delta * len(percpu)` otherwise you cannot guarantee that you'll get a floating point division between `cpu_delta` and `system_delta` (if those happen to be integers at that particular point in time).

Reply:
I didn't notice that. Changed it because it gave flake8 errors. Changed to:
               ```pre_system_cpu_usage = precpu_stats['system_cpu_usage']
                system_delta = float(system_cpu_usage - pre_system_cpu_usage)
                if system_delta > 0 and cpu_delta > 0:
                    cpu_percent = cpu_delta / system_delta * len(percpu) 
                    cpu_percent *= 100```